doc/api/http.md

```diff
@@ -625,8 +625,7 @@ added: v0.5.9
 * `timeout` {number} Milliseconds before a request times out.
 * `callback` {Function} Optional function to be called when a timeout occurs. Same as binding to the `timeout` event.
 
-Once a socket is assigned to this request and is connected
-[`socket.setTimeout()`][] will be called.
+Once a socket is assigned to this request [`socket.setTimeout()`][] will be called.
```

Comment:
This is not 100% accurate.  Even on 8.x `socket.setTimeout()` will be called when `'connect'` is emitted on the `net.Socket` if `request.setTimeout()` is called after the `'socket'` event is emitted on the `http.ClientRequest` object but before the `'connect'` event is emitted on the `net.Socket`

Reply:
True!

How about:

> If no socket is assigned to this request then [`socket.setTimeout()`][] will be called immediately. Otherwise [`socket.setTimeout()`][] will be called after the assigned socket is connected.

Reply:
Sounds good.